cvmfs/cvmfs_config

```diff
@@ -1624,6 +1627,131 @@ runas() {
+  done
+}
+
+__usrlocal_migrate_clean_legacy_directories() {
+  echo "cleaning up legacy directories..."
+  for directory in /etc/bash_completion.d/cvmfs \
+                   /usr/share/doc/cvmfs-*; do
+    if [ -d $directory ]; then
+      rm -fR "$directory"
+      echo "found and removed legacy directory '$directory'"
+    fi
+  done
+}
+
+__usrlocal_migrate_clean_and_rescue_legacy_configuration() {
+  local old_conf_dir="/etc/cvmfs"
+  local new_conf_dir="/usr/local/etc/cvmfs"
+
+  echo "cleaning up legacy configuration..."
+  if [ -L $old_conf_dir ] &&
+     [ x"$(readlink $old_conf_dir)" = x"$new_conf_dir" ]; then
+    echo "configuration already points to $new_conf_dir. do nothing."
+    return 0
+  fi
+
+  if [ ! -e $old_conf_dir ]; then
+    echo "no legacy configuration found. do nothing."
+    return 0
+  fi
+
+  # detect and remove all configuration files of previous CernVM-FS installs
+  # Note: This deletes only the configuration files that have been shipped with
+  #       CernVM-FS. Everything user-added stays intact.
+  for conf_file in config.d/atlas-nightlies.cern.ch.conf            \
+                   config.d/cms.cern.ch.conf                        \
+                   config.d/grid.cern.ch.conf                       \
+                   config.sh                                        \
+                   default.conf                                     \
+                   default.d/50-cern.conf                           \
+                   default.d/60-egi.conf                            \
+                   default.d/README                                 \
+                   domain.d/cern.ch.conf                            \
+                   domain.d/egi.eu.conf                             \
+                   domain.d/opensciencegrid.org.conf                \
+                   keys/cern.ch/cern-it1.cern.ch.pub                \
+                   keys/cern.ch/cern-it2.cern.ch.pub                \
+                   keys/cern.ch/cern-it3.cern.ch.pub                \
+                   keys/cern.ch/cern.ch.pub                         \
+                   keys/egi.eu/egi.eu.pub                           \
+                   keys/opensciencegrid.org/opensciencegrid.org.pub \
+                   serverorder.sh; do
+    local old_conf_file="${old_conf_dir}/${conf_file}"
+    if [ -f $old_conf_file ]; then
+      local quarantined_conf_file="${new_conf_dir}/quarantine/${conf_file}"
+      mkdir -p "$(dirname $quarantined_conf_file)"
+      mv -f "$old_conf_file" "$quarantined_conf_file"
+      echo "found and quarantined legacy configuration file '$conf_file'"
+    fi
+  done
+
+  # rescue all configuration files that have been added by the user
+  local cwd="$(pwd)"
+  cd "$old_conf_dir" # remove $old_conf_dir prefix from `find` output
+  for conf_file in $(find . -type f); do
+    echo "found and rescued user added configuration file '$conf_file'"
+    mv -f $conf_file ${new_conf_dir}/${conf_file}
```

Comment:
Here it might be necessary to create the base directory in `${new_conf_dir}` first